contrib/slime-cl-indent.el

```diff
@@ -1116,14 +1120,22 @@ environment\\|more\
                           (+ col
                              lisp-lambda-list-keyword-parameter-indentation)
                         col))))
-              (if (looking-at lisp-indent-lambda-list-keywords-regexp)
-                  (setq indent
-                        (if lisp-lambda-list-keyword-parameter-alignment
-                            (or indent pos)
-                          (+ col
-                             lisp-lambda-list-keyword-parameter-indentation))
-                        next nil)
-                (setq indent col))))
+              (cond
+               ((looking-at lisp-indent-lambda-list-single-arg-keywords-regexp)
+                ;; Some keywords such as &whole have a single argument;
+                ;; following arguments are indented to the beginning of the
+                ;; lambda-list.
+                (setq indent col
+                      next nil))
+               ((looking-at lisp-indent-lambda-list-keywords-regexp)
```

Comment:
There's another reference to `lisp-indent-lambda-list-keywords-regexp` earlier in this function. Does it need to be adapted too?

Reply:
I do not think so; I tried lots of adjustments because the function is really opaque, and this is how I got indentation which works as intended in without affecting other lambda lists.

I'm pretty sure that there are other edge cases (e.g. the fact that &ALLOW-OTHER-KEYS does not take any argument), but at this point I am just trying to fix a very specific problem without breaking anything.

Reply:
ACK. Sounds reasonable.